azurerm/internal/services/analysisservices/analysis_services_server_resource.go

```diff
@@ -252,6 +252,32 @@ func resourceArmAnalysisServicesServerUpdate(d *schema.ResourceData, meta interf
 		return err
 	}
 
+	serverResp, err := client.GetDetails(ctx, id.ResourceGroup, id.Name)
+	if err != nil {
+		if utils.ResponseWasNotFound(serverResp.Response) {
+			d.SetId("")
+			return nil
```

Comment:
this is the update - so if this is a 404 we should error (since the Read function will be run prior to this - so this'd be changes made OOB during the run)

Reply:
Done